### PR TITLE
Stop updating shard range in citus_update_shard_statistics for append tables

### PR DESCRIPTION
**Pull request by marcocitus**
DESCRIPTION: Stop updating shard range in citus_update_shard_statistics for append tables

We want to get to a point where append tables allow inserts into arbitrary shards to enable horizontal write scalability (#5358). That means we should not have an out-of-band process setting shardminvalue/shardmaxvalue, because that would cause shard pruning to exclude these shards for some queries, and we cannot guarantee that there will not be additional writes that fall outside the range. 

We should at some point build a mechanism for marking shards as read-only, at which point it is safe to update these ranges. However, that will likely be separate mechanism from citus_update_shard_statistics. Alternatively, append tables might not have a distribution column at all.

I removed the citus_update_table_statistics isolation test which was not realy showing anything meaningful (since we unlock the table before the end of the transaction, drop table is not really expected to block) and suffering from a race condition that got triggered by citus_update_table_statistics becoming a tiny bit faster.

### CODE REVIEW

File: src/test/regress/expected/citus_update_table_statistics.out [hunk 1]
```diff
@@ -91,25 +91,9 @@ SELECT
 FROM pg_dist_shard ds JOIN pg_dist_shard_placement dsp USING (shardid)
 WHERE ds.logicalrelid::regclass::text in ('test_table_statistics_hash') AND dsp.shardlength > 0
 ORDER BY 2, 3;
-         tablename          | shardid | placementid |             shardname             | shardminvalue | shardmaxvalue
+ tablename | shardid | placementid | shardname | shardminvalue | shardmaxvalue
```
Comment (SaitTalhaNisanci) [inline src/test/regress/expected/citus_update_table_statistics.out:94]: What are we testing with this now?

File: src/test/regress/output/multi_copy.source [hunk 2]
```diff
@@ -25,24 +25,24 @@ ERROR:  could not find any shards into which to copy
...
+---------------------------------------------------------------------
+
 (1 row)
 
 -- Test empty copy
 COPY customer_copy_hash FROM STDIN;
 -- Test syntax error
 COPY customer_copy_hash (c_custkey,c_name) FROM STDIN;
-ERROR:  invalid input syntax for type integer: "1,customer1"
+ERROR:  invalid input syntax for integer: "1,customer1"
```
Comment (SaitTalhaNisanci) [inline src/test/regress/output/multi_copy.source:37]: why does this error change?

**Review by SaitTalhaNisanci — COMMENTED** [review]
It seems like we are reverting most of the additions of https://github.com/citusdata/citus/pull/4752/files right?

File: src/test/regress/output/multi_copy.source (hunk 2 above)
Comment (marcocitus) [reply src/test/regress/output/multi_copy.source:37]: I think it changed in a major Postgres version, and the output differences are ignored.

**Comment by marcocitus** [comment]
> It seems like we are reverting most of the additions of #4752 (files) right?

not the important ones, deadlock detection is still there

**Comment by SaitTalhaNisanci** [comment]
>not the important ones, deadlock detection is still there

Yeah, I miswrote the sentence I guess, what I meant was most of the things we are removing with this PR were added with that one

File: src/backend/distributed/metadata/metadata_utility.c [hunk 3]
```diff
@@ -497,12 +481,12 @@ ReceiveShardNameAndSizeResults(List *connectionList, Tuplestorestate *tupleStore
 			memset(values, 0, sizeof(values));
 			memset(isNulls, false, sizeof(isNulls));
 
-			char *tableName = PQgetvalue(result, rowIndex, 0);
+			char *tableName = PQgetvalue(result, rowIndex, 1);
```
Comment (SaitTalhaNisanci) [inline src/backend/distributed/metadata/metadata_utility.c:484]: I guess `0` was the shard id

File: src/backend/distributed/metadata/metadata_utility.c [hunk 4]
```diff
@@ -881,12 +865,10 @@ GetSizeQueryBySizeQueryType(SizeQueryType sizeQueryType)
 
 /*
  * GenerateAllShardStatisticsQueryForNode generates a query that returns:
- * - all shard_name, shard_size pairs for the given node (if useShardMinMaxQuery is false)
- * - all shard_id, shard_minvalue, shard_maxvalue, shard_size quartuples (if true)
+ * all shard_name, shard_size pairs for the given node
```
Comment (SaitTalhaNisanci) [inline src/backend/distributed/metadata/metadata_utility.c:868]: Do we return just `shard_name, shard_size` pair or `shard_id, shard_name, shard_size` tuple? It seems like the latter

File: src/backend/distributed/metadata/metadata_utility.c [hunk 5]
```diff
@@ -904,61 +886,32 @@ GenerateAllShardStatisticsQueryForNode(WorkerNode *workerNode, List *citusTableI
...
 	return allShardStatisticsQuery->data;
 }
 
 
 /*
- * GenerateShardStatisticsQueryForShardList generates one of the two types of queries:
- * - SELECT shard_name - shard_size (if useShardMinMaxQuery is false)
- * - SELECT shard_id, shard_minvalue, shard_maxvalue, shard_size (if true)
+ * GenerateShardStatisticsQueryForShardList generates a query that returns:
+ * SELECT shard_name, shard_size
```
Comment (SaitTalhaNisanci) [inline src/backend/distributed/metadata/metadata_utility.c:904]: Again here should it be `SELECT shard_id, shard_name, shard_size`

File: src/backend/distributed/metadata/metadata_utility.c (hunk 3 above)
Comment (marcocitus) [reply src/backend/distributed/metadata/metadata_utility.c:467]: added a comment

**Review by SaitTalhaNisanci — APPROVED** [review]

**Pull request closed without merging**